### PR TITLE
haskellPackages: Assorted build fixes

### PR DESCRIPTION
**Pull request by rvl**
This is for merging into the `haskell-updates` branch (PR #339272).

Tested the build with:
```
nix build --impure --expr 'let pkgs = (import ./. { system = "x86_64-linux"; });
in pkgs.haskellPackages.ghcWithHoogle (ps: with ps; [
  gi-gtk-declarative
  gi-gtk-declarative-app-simple
  reflex-gloss
  reflex-gi-gtk
  reflex-sdl2
  pipes-extras
  pipes-http
  vivid
  dear-imgui
])'
```

Upstream PRs:
- owickstrom/gi-gtk-declarative#118
- Gabriella439/Haskell-Pipes-Extras-Library#19
- Gabriella439/Haskell-Pipes-HTTP-Library#17
- merijn/broadcast-chan#19

### CODE REVIEW

File: pkgs/development/haskell-modules/configuration-common.nix [hunk 1]
```diff
@@ -1912,6 +1918,9 @@ self: super: {
   # https://github.com/Gabriella439/Haskell-Pipes-HTTP-Library/pull/17
   pipes-http = doJailbreak super.pipes-http;
 
+  # 2024-09-18: transformers <0.6
+  pipes-extras = doJailbreak super.pipes-extras;
```
Comment (ysangkok) [inline pkgs/development/haskell-modules/configuration-common.nix:1922]: Why is this necessary? It was fixed in revision 7: https://hackage.haskell.org/package/pipes-extras-1.0.15/revisions/

Comment (rvl) [reply pkgs/development/haskell-modules/configuration-common.nix:1922]: I couldn't figure out why! In `hackage-packages.nix`, pipes-extras has `revision = "8"`. In any case, adding a version assertion is a good idea.

Comment (rvl) [reply pkgs/development/haskell-modules/configuration-common.nix:1922]: Aha. The test-suite component still has `transformers <0.6`.

https://hackage.haskell.org/package/pipes-extras-1.0.15/revision/8.cabal

**Comment by maralorn** [comment]
This is amazingly thorough. Thank you so much.

I was just confused by vivid missing a link until I read the commit message. Would you mind putting that message into a comment anyway? Then the next time someone tries to figure out whether they can drop the override they don’t go looking for the repo.

**Comment by rvl** [comment]
> Would you mind putting that message into a comment anyway?

OK, done, thanks @maralorn.

Also while looking for the author's e-mail address, I noticed that [vivid-0.5.2.1](https://hackage.haskell.org/package/vivid-0.5.2.1) was released about a week ago, which fixes the compile error.

**Comment by maralorn** [comment]
Excellent!
